src/containerapp/azext_containerapp/containerapp_env_certificate_decorator.py

```diff
@@ -191,3 +193,36 @@ def list(self):
         managed_certs = self.get_managed_certificates(certificate_name)
         private_certs = self.get_private_certificates(certificate_name)
         return managed_certs + private_certs
+
+class ContainerappEnvCertificatePreviweUploadDecorator(ContainerappEnvCertificateUploadDecorator):
+    def validate_arguments(self):
+        # validate arguments
+        if self.get_argument_certificate_file() and self.get_argument_certificate_key_vault_url():
+            raise ValidationError("Cannot use certificate file/password with certificate identity/keyvaulturl at the same time")
+        if (not self.get_argument_certificate_file()) and (not self.get_argument_certificate_key_vault_url()):
+            raise ValidationError("Either certificate file/password or certificate identity/keyvaulturl should be set when hostName is set")
```

Comment:
Please use argument name in message, for example:
`--custom-domain-certificate-key-vault-url/--certificate-key-vault-url`